src/Raven.Client/Documents/Session/SessionTimeSeriesBase.cs

```diff
@@ -122,6 +122,31 @@ public void Delete(DateTime? from = null, DateTime? to = null)
             {
                 Session.Defer(new TimeSeriesBatchCommandData(DocId, Name, appends: null, deletes: new List<TimeSeriesOperation.DeleteOperation> { op }));
             }
+
+            RemoveFromCacheIfNeeded(from, to);
+        }
+
+        private void RemoveFromCacheIfNeeded(DateTime? from = null, DateTime? to = null)
+        {
+            if (Session.TimeSeriesByDocId.TryGetValue(DocId, out var cache))
+            {
+                if (from == null || from == DateTime.MinValue && to == null || to == DateTime.MaxValue)
+                {
+                    cache.Remove(Name);
+                }
+                else if (cache.TryGetValue(Name, out var ranges) && ranges.Count > 0)
+                {
+                    var toDelete = new List<int>();
+                    for (var i = 0; i < ranges.Count; i++)
+                    {
+                        if (ranges[i].From <= from && ranges[i].To >= to)
```

Comment:
What is the output of `DateTime.Now <= null` ?

Reply:
done :)